kartothek/io/iter.py

```diff
@@ -159,6 +167,14 @@ def read_dataset_as_dataframes__iterator(
             {'core': pd.DataFrame, 'lookup': pd.DataFrame}
         ]
     """
+    if tables is None and isinstance(columns, dict):
```

Comment:
I think we can remove this and also use `if len(ds_factory.tables) > 1:` here, to make sure we always raise the warning if a user reads a multi-tabled dataset.
(You'll have to define `ds_factory = _ensure_factory(...)` before)

Reply:
Okay will define the `ds_factory` and make the changes.